conda_forge_tick/feedstock_parser.py

```diff
@@ -0,0 +1,346 @@
+
+
+def _extract_requirements(meta_yaml):
+    strong_exports = False
+    requirements_dict = defaultdict(set)
+    for block in [meta_yaml] + meta_yaml.get("outputs", []) or []:
+        req: "RequirementsTypedDict" = block.get("requirements", {}) or {}
+        if isinstance(req, list):
+            requirements_dict["run"].update(set(req))
+            continue
+        for section in ["build", "host", "run"]:
+            requirements_dict[section].update(
+                list(as_iterable(req.get(section, []) or [])),
+            )
+        test: "TestTypedDict" = block.get("test", {})
+        requirements_dict["test"].update(test.get("requirements", []) or [])
+        requirements_dict["test"].update(test.get("requires", []) or [])
+        run_exports = (block.get("build", {}) or {}).get("run_exports", {})
+        if isinstance(run_exports, dict) and run_exports.get("strong"):
+            strong_exports = True
+    for k in list(requirements_dict.keys()):
+        requirements_dict[k] = {v for v in requirements_dict[k] if v}
+    req_no_pins = {
+        k: {PIN_SEP_PAT.split(x)[0].lower() for x in v}
+        for k, v in dict(requirements_dict).items()
+    }
+    return dict(requirements_dict), req_no_pins, strong_exports
+
+
+def _fetch_static_repo(name, dest):
+    r = requests.get(
+        f"https://github.com/conda-forge/{name}-feedstock/archive/master.zip",
+    )
+    if r.status_code != 200:
+        LOGGER.error(
+            f"Something odd happened when fetching feedstock {name}: {r.status_code}",
+        )
+        return r
+
+    zname = os.path.join(dest, f"{name}-feedstock-master.zip")
+
+    with open(zname, "wb") as fp:
+        fp.write(r.content)
+
+    z = zipfile.ZipFile(zname)
+    z.extractall(path=dest)
+    dest_dir = os.path.join(dest, os.path.split(z.namelist()[0])[0])
+    return dest_dir
+
+
+def populate_feedstock_attributes(
+    name: str,
+    sub_graph: typing.MutableMapping,
+    meta_yaml: typing.Union[str, Response] = "",
+    conda_forge_yaml: typing.Union[str, Response] = "",
+    mark_not_archived=False,
+    feedstock_dir=None,
+) -> typing.MutableMapping:
+    """Parse the various configuration information into something usable
+
+    Notes
+    -----
+    If the return is bad hand the response itself in so that it can be parsed
+    for meaning.
+    """
+    sub_graph.update({"feedstock_name": name, "bad": False, "branch": "master"})
+
+    if mark_not_archived:
+        sub_graph.update({"archived": False})
+
+    # handle all the raw strings
+    if isinstance(meta_yaml, Response):
+        sub_graph["bad"] = f"make_graph: {meta_yaml.status_code}"
+        return sub_graph
+    sub_graph["raw_meta_yaml"] = meta_yaml
+
+    # Get the conda-forge.yml
+    if isinstance(conda_forge_yaml, str):
+        sub_graph["conda-forge.yml"] = {
+            k: v
+            for k, v in yaml.safe_load(conda_forge_yaml).items()
+            if k
+            in {
```

Comment:
This should most likely be pulled up into a constant, and then eventually out into smithy?

Reply:
This the list of keys we keep from the conda-forge.yml. Happy to put it as a constant but it should not be in smithy. We don't keep all of the keys, likely due to it having had encrypted secrets in it.

Reply:
Moved to a global.